src/main/PurchaseArray.java

```diff
@@ -0,0 +1,54 @@
+package main;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Comparator;
+
+
+public class PurchaseArray {
+    private final AbstractPurchase[] purchases;
+
+    public PurchaseArray(AbstractPurchase[] purchases) {
+        this.purchases = purchases;
+    }
+
+    public void printArr(){
+        for(int i = 0; i < purchases.length; i++) {
+            System.out.println(purchases[i]);
+        }
+    }
+    public void printByIndex(int index) {
+        System.out.println(purchases[index]);
+    }
+
+    public void sort(){
+        Arrays.sort(purchases);
+    }
+
+    public String getMinimalCost(){
+        // new array, so the original doesn't gets sorted
+        AbstractPurchase[] array = purchases.clone();
```

Comment:
```suggestion
        AbstractPurchase[] array = purchases.copy();
```
[Avoid clone](http://www.javapractices.com/topic/TopicAction.do?Id=71)